docs/src/starks/builtins.md

```diff
@@ -0,0 +1,27 @@
+# Builtins
+
+We can understand the built-in as a small machine, that we can use to efficiently prove a subprogram. For example, it may be able to prove a hash, like Poseidon or Keccak, verify a signature, or check that some variable is in a range, and the cost would be less than what we would have if using the Cairo VM instructions.
+
+For each subprogram we want to prove, we will have a machine, which will have its own set of constraints in the prover. Let's take for example the Range Check built-in. This builtin enforces that a value $X$ is between 0 and 2^128.
+
+The logic behind the built-in is pretty straightforward. We split $X$ into 8 parts. So we will say that $X = X_{0} + X_{1} * 2^16 + X_{2} * 2^32 + X_{3} * 2^48 + ... + X_{7} * 2^112$
+
+Then we require that each is in the range $ 0 < X_{i} < 2^16 $. The idea here is to reuse the Range Check constraint that checks if the offsets are between $0$ and $2^16$. If we can decompose the number in eight limbs of 16 bits, and we don't need any more limbs, it follows that the number will be less than $2^128$
```

Comment:
I think Github is not rendering properly this piece of latex:

`$ 0 < X_{i} < 2^16 $`

I think probably the whitespaces need to be removed at the beginning and at the end:

`$0 < X_{i} < 2^16$`